File: src/data/courses/dsa/string-hashing-problems/hashing-based-problems/contains-duplicate/code/python/contains_duplicate_sol2.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # Sort + Scan: O(n log n) — Duplicates become adjacent
        nums.sort()
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]: return True
        return False

File: src/data/courses/dsa/string-hashing-problems/hashing-based-problems/contains-duplicate/code/java/ContainsDuplicateSol2.java
class Solution {
    // Sort + Scan: O(n log n) — Duplicates become adjacent
    public boolean containsDuplicate(int[] nums) {
        Arrays.sort(nums);
        for (int i = 1; i < nums.length; i++)
            if (nums[i] == nums[i - 1]) return true;
        return false;
    }
}